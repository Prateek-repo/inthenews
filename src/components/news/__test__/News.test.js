import { shallow } from "enzyme";
import "../../../setupTests";
import toJson from "enzyme-to-json";
import Container from "../News";
import fetchMock from "jest-fetch-mock";

describe("<News/>", () => {
  let wrapper;

  const defaultProps = {
    progressBar: jest.fn(),
  };

  beforeEach(() => {
    wrapper = shallow(<Container {...defaultProps} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("SnapShot creation", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("handling componentDidUpdate", () => {
    wrapper.setProps({
      category: "",
    });

    const prevProps = {
      country: "",
    };

    const componentInstance = wrapper.instance();
    jest.spyOn(componentInstance, "getTheNews");
    componentInstance.componentDidUpdate(prevProps);
    expect(componentInstance.getTheNews).toHaveBeenCalledTimes(1);
  });

  it("handling fetchMoreData", () => {
    wrapper.setState({
      page: 1,
    });

    const componentInstance = wrapper.instance();
    componentInstance.fetchMoreData();
    expect(componentInstance.state.page).toEqual(2);
  });

  it("Rendering the News Component successfully", () => {
    wrapper.setProps({
      darkMode: true,
      category: "",
    });
    wrapper.setState({
      loading: false,
    });
    expect(wrapper).toBeTruthy();
  });

  it("Return score", async () => {
    wrapper.setState({
      articles: [{ n: "" }, { n: "" }],
    });
    fetchMock.resetMocks();
    fetchMock.mockResponseOnce(
      JSON.stringify({
        result: [
          {
            title: "JSpace Rocket",
            description: "",
          },
        ],
      })
    );
    await wrapper.instance().getMoreNews();
    expect(wrapper.instance().state.loading).toEqual(false);
  });
});
