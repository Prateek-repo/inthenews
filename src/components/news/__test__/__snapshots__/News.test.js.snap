// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<News/> SnapShot creation 1`] = `
<Fragment>
  <h2
    className="text-center"
    style={
      Object {
        "color": null,
        "marginTop": "56px",
      }
    }
  >
    Top Headlines:
     
    General
  </h2>
  <div>
    <Spinner />
  </div>
</Fragment>
`;
